lib/buffer.js

```diff
@@ -899,6 +899,10 @@ Buffer.prototype.compare = function compare(target,
 // - encoding - an optional encoding, relevant if val is a string
 // - dir - true for indexOf, false for lastIndexOf
 function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
+  if (!Buffer.isBuffer(buffer)) {
```

Comment:
We generally have moved to accepting all `ArrayBufferView`s, or at least `Uint8Array`s where other `ArrayBufferView`s don’t make sense, in places where `Buffer`s are accepted.

```suggestion
  if (!isArrayBufferView(buffer)) {
```

Reply:
I think at least a `DataView` is not allowed because of the missing length property. In the end this function is called via `Buffer` member functions so anything else then a `Buffer` is suspicious.
Will take a look.

Reply:
@flarna We did intentionally move all `Buffer` functions towards working on any `Uint8Array`, since `Buffer` is just `Uint8Array` with a few extra methods.

Regarding the missing `length` property, that’s true, but if we use `.byteLength` I think this should work as expected?

Reply:
Any `ArrayBufferView` works now.